omniscidb/QueryEngine/Execute.cpp

```diff
@@ -2790,66 +2790,109 @@ std::vector<std::unique_ptr<ExecutionKernel>> Executor::createHeterogeneousKerne
 
   CHECK(!ra_exe_unit.input_descs.empty());
 
+  const bool use_multifrag_kernel = eo.allow_multifrag && is_agg;
```

Comment:
This reads as group-bys would not use a multi-fragment policy. Needs cleanup I guess.